Remotion/Web/ClientScript/Scripts/Utilities.ts

```diff
@@ -0,0 +1,335 @@
+  };
+
+  public ExecuteResizeHandlers(): void
+  {
+    var existingResizeHandlers = new Array();
+    for (var i = 0; i < this._resizeHandlers.length; i++)
+    {
+      var item = this._resizeHandlers[i];
+      var element = $ (item.Selector);
+      if (element != null && element.length > 0)
+      {
+        item.Handler (element);
+        existingResizeHandlers[existingResizeHandlers.length] = item;
+      }
+    }
+    this._resizeHandlers = existingResizeHandlers;
+  };
+
+  public IsInDom (element: HTMLElement): boolean
+  {
+    ArgumentUtility.CheckNotNull('element', element);
+
+    var html = window.document.body.parentNode;
+    let node: Nullable<Node & ParentNode> = element;
+    while (node)
+    {
+      if (node === html)
+          return true;
+      node = node.parentNode;
+    }
+    return false;
+  }
+}
+
+type PageUtility_ResizeHandler = (element: JQuery) => void;
+
+class PageUtility_ResizeHandlerItem
+{
+  public readonly Selector: string;
+  public Handler;
+
+  constructor(selector: string, handler: PageUtility_ResizeHandler)
+  {
+    this.Selector = selector;
+    this.Handler = handler;
+  }
+}
+
+PageUtility.Instance = new PageUtility();
+
+class WebServiceUtility
+{
+  static Execute<T>(serviceUrl: string, serviceMethod: string, params: object, onSuccess: (result: T) => void, onError: (err: Sys.Net.WebServiceError) => void): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsString ('serviceUrl', serviceUrl);
+    ArgumentUtility.CheckNotNullAndTypeIsString ('serviceMethod', serviceMethod);
+    ArgumentUtility.CheckNotNullAndTypeIsObject ('params', params);
+    ArgumentUtility.CheckNotNullAndTypeIsFunction ('onSuccess', onSuccess);
+    ArgumentUtility.CheckNotNullAndTypeIsFunction ('onError', onError);
+
+    if (Sys === undefined)
+      throw "'Sys' namespace is undefined. System.Web.ScriptManager has not been included on the page.";
+
+    // re-motion: replaced jQuery AJAX call with .NET call because of the following problem:
+    //           when extending the parameter list with the necessary arguments for the web service method call,
+    //           the JSON object is serialized to "key=value;" format, but the service expects JSON format ("{ key: value, ... }")
+    //           see http://encosia.com/2008/06/05/3-mistakes-to-avoid-when-using-jquery-with-aspnet-ajax/ 
+    //           under "JSON, objects, and strings: oh my!" for details.
+
+    let executingRequest: Nullable<Sys.Net.WebRequest> = Sys.Net.WebServiceProxy.invoke (
+      serviceUrl,
+      serviceMethod,
+      false,
+      params,
+      function (result: string) {
+        executingRequest = null;
+        onSuccess (result);
+      },
+      function (err: Sys.Net.WebServiceError) {
+        executingRequest = null;
+        var isTimedOut = err.get_timedOut();
+        var isAborting = !isTimedOut && err.get_statusCode() === -1;
+        if (!isAborting)
+        {
+          onError (err);
+        }
+      });
+  };
```

Comment:
onSuccess and onError arguments in JS disagree with official type declarations. The types from the actual source code win.

```
if ((typeof(statusCode) !== "undefined") && (statusCode !== 200)) {
                if (onFailure) {
                    error = new Sys.Net.WebServiceError(false,
                            data.Message || String.format(Sys.Res.webServiceFailedNoMsg, methodName),
                            data.StackTrace || null,
                            data.ExceptionType || null,
                            data);
                    error._statusCode = statusCode;
                    onFailure(error, userContext, methodName);
                }
                else {
                    if (data.StackTrace && data.Message) {
                        error = data.StackTrace + "-- " + data.Message;
                    }
                    else {
                        error = data.StackTrace || data.Message;
                    }
                    error = String.format(error ? Sys.Res.webServiceFailed : Sys.Res.webServiceFailedNoMsg, methodName, error);
                    throw Sys.Net.WebServiceProxy._createFailedError(methodName, String.format(Sys.Res.webServiceFailed, methodName, error));
                }
            }
            else if (onSuccess) {
                onSuccess(data, userContext, methodName);
            }
```